pyccel/ast/bitwise_operators.py

```diff
@@ -41,16 +41,18 @@ class PyccelInvert(PyccelUnaryOperator):
     """
     __slots__ = ()
     _precedence = 14
+    _dtype     = NativeInteger()
 
-    def _set_dtype(self):
-        self._dtype     = NativeInteger()
-        a = self._args[0]
+    def _calculate_dtype(self, *_args):
+        _dtype = NativeInteger()
```

Comment:
You should either have a class variable which you return:
```
class PyccelInvert(..):
    ...
    _dtype = NativeInteger()
    
    def _calculate_dtype(self, *_args):
         dtype = self._dtype
         ....
         return dtype, _precision
```
Or you need to remove the class variable

Reply:
done [f29d118](https://github.com/pyccel/pyccel/pull/797/commits/f29d1184964b4a001e57e7ad97cebec75c31d550)

Reply:
You still need to specify `self` to access this variable. See the pylint error